packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -611,24 +611,7 @@ class FlutterPlugin implements Plugin<Project> {
                     with flutterTask.assets
                 }
             }
-            if (packageAssets) {
-                String mainModuleName = "app"
-                try {
-                    String tmpModuleName = project.rootProject.ext.mainModuleName
```

Comment:
These days many many projects have a custom name for the app module, actually, many projects have multiple app modules.
So hardcoding the `:app` completely breaks many projects.

by removing this script, that makes this [issue](https://github.com/flutter/flutter/pull/27154) come back again.

We need a way to let developers set their app module name, or it will be better if you can detect it automatically in this script.

Please let me know if I can help!
Thank you!
@blasten @zanderso @xster

Reply:
Potentially, we could iterate through all the subprojects and find the one that applies the `com.android.application` plugin. Is this something you would like to explore?

Reply:
Sure, I'd figure out that!